sha1solver/sha1solver.c

```diff
@@ -0,0 +1,141 @@
+#include "sha1.h"
+#include "stdio.h"
+#include "string.h"
+#include "stdlib.h"
+
+#define hex(c) ((*(c)>='a')?*(c)-'a'+10:(*(c)>='A')?*(c)-'A'+10:*(c)-'0') 
+
+int SHA1Progress(
+    char *hash_out,
+    char *str,
+    int len,
+    int poz_start,
+    int poz_end,
+    SHA1_CTX * ctx,
+    char *pattern,
+    int pattern_len,
+    int r,
+    int s,
+    int first)
+{
+    unsigned int ii;
```

Comment:
Consider moving this declaration to loop initialization statement. That will make clear for each loop this variable is used. Note that you are doing it somewhere anyway.

Reply:
done, but that is obvious for one loop.